ivadomed/loader/loader.py

```diff
@@ -424,6 +424,83 @@ def get_pair_slice(self, slice_index, gt_type="segmentation"):
 
         return dreturn
 
+    def read_file(self, filename):
+        """Read file according to file type.
+        Args:
+            filename (str): Subject filename.
+        """
+        if "nii" in self.extension:
+            # Returns 'nibabel.nifti1.Nifti1Image' object
+            # For 'nii' and 'nii.gz' extentions
+            return nib.load(filename)
+        elif self.extension == "png":
+            # Returns data from file as a 3D numpy array
+            # Behavior for grayscale png only, behavior TBD for RGB or RBGA png
+            return np.expand_dims(imageio.imread(filename, as_gray=True), axis=-1)
+        else:
+            raise RuntimeError('Input file type not supported')
+
+    def apply_canonical(self, data):
+        """Apply nibabel as_closest_canonical function to nifti data only.
+        Args:
+            data ('nibabel.nifti1.Nifti1Image' object or 'ndarray'):
+                for nifti or png file respectively.
+        """
+        if "nii" in self.extension:
+            # Returns 'nibabel.nifti1.Nifti1Image' object
+            return nib.as_closest_canonical(data)
+        if self.extension == "png":
+            # Returns data as is in numpy array
+            return data
```

Comment:
Instead of checking `self.extension`, it might be more reliable to check the type of the `data` variable, since the behavior is specific to `Nifti1Image` and `ndarray` types.

(This also applies to other functions here.)

Reply:
Also, two `if` statements are used. What about the `if .. elif .. else raise` pattern like was done in `read_file`?

Reply:
I kept the condition on the file extension for now. Because we use BIDS convention, the accepted file formats are limited. For microscopy, we will also want a loader for OMETIFF (not implemented yet), and the behavior with ndarray may differ in some cases (ex: voxel size could be extract from file metadata instead of bids metadata).

I corrected the `if/elif` statements but I did not include an `else` in the functions because `read_file` is the first to be called with the same `if/elif` conditions and would raise the error. Is it bad practice? It would be necessary to add an `else` if we change de `if/elif` condition though.

Reply:
> For microscopy, we will also want a loader for OMETIFF (not implemented yet), and the behavior with ndarray may differ in some cases (ex: voxel size could be extract from file metadata instead of bids metadata).

Thank you for the clarification. Apologies for making that suggestion without having the full context. :sheep: 

> I corrected the if/elif statements (in d9124b2) but I did not include an else in the functions because read_file is the first to be called with the same if/elif conditions and would raise the error. Is it bad practice?

Good point. I was thinking "what if `read_file` isn't called first?" but I guess there's not much of a risk of that happening, is there...

Still, perhaps we could raise that error in `__init__` when `self.extension` is first set. That way we guarantee the error is thrown before anything else happens (when `loader` is first created).

Reply:
> Still, perhaps we could raise that error in `__init__` when `self.extension` is first set. That way we guarantee the error is thrown before anything else happens (i.e. when `loader` is first created).

Fixed in 66b2e90